bokeh/charts/_data_adapter.py

```diff
@@ -60,7 +63,14 @@ def __init__(self, data, index=None, columns=None, force_alias=True):
                 columns = list(keys())
 
             elif keys is None:
-                columns = list(map(str, range(len(data))))
+                if blaze and isinstance(self._values, blaze.interactive.InteractiveSymbol):
+                    keys = self._values.fields
+                    if len(keys) == len(self._values.data):
+                        columns = keys
```

Comment:
What is the intent of this check?  You probably don't want to interact with the `data` field because that could vary wildly.  It might be a sqlalchemy.Table or a DataFrame or ....  We can't guarantee that these define `__len__` to all mean the same thing.